src/SourceBuild/content/eng/pipelines/ci.yml

```diff
@@ -1,55 +1,24 @@
-# This is the main build definition (PR+CI) for dotnet/dotnet
-
-trigger:
-  batch: true
-  branches:
-    include:
-    - main
-    - release/*
-    - internal/release/*
-
-pr:
-  branches:
-    include:
-    - main
-    - release/*
-    - internal/release/*
-
-stages:
-- ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
-  - template: templates/stages/vmr-scan.yml
-
-  # For rolling builds we want to build the MSFT SDK first so that we can
-  # compare the contents with the source-built one later.
-  # This only works because we don't run this test in PRs. If we decided
-  # to run this test in PRs, we would need to build the MSFT SDK there too.
-  - stage: Build_MSFT_SDK
-    displayName: Build MSFT SDK
-    dependsOn: []
-    jobs:
-    - template: /src/installer/eng/build.yml
-      parameters:
-        agentOs: Linux
-        jobName: Build_LinuxPortable_Release_x64
-        buildConfiguration: Release
-        buildArchitecture: x64
-        linuxPortable: true
-        runTests: false
-        isBuiltFromVmr: true
-    - template: /src/installer/eng/build.yml
-      parameters:
-        agentOs: Linux
-        jobName: Build_Arm64_Release
-        buildConfiguration: Release
-        buildArchitecture: arm64
-        runtimeIdentifier: 'linux-arm64'
-        linuxPortable: true
-        runTests: false
-        isBuiltFromVmr: true
-
-- template: /src/installer/eng/pipelines/templates/stages/vmr-build.yml
-  parameters:
-    isBuiltFromVmr: true
-    ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
-      dependsOn: Build_MSFT_SDK
-      condition: always()
+# This is the main build definition (PR+CI) for dotnet/dotnet
+
+trigger:
+  batch: true
+  branches:
+    include:
+    - main
+    - release/*
+    - internal/release/*
+
+pr:
+  branches:
+    include:
+    - main
+    - release/*
+    - internal/release/*
+
+stages:
+- ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+  - template: templates/stages/vmr-scan.yml
+
+- template: /src/installer/eng/pipelines/templates/stages/vmr-build.yml

```

Comment:
Please double check this but I believe that this template checks for `isBuildFromVmr` and adds the `dependsOn` to depend on the `Build_MSFT_SDK` stage

Reply:
the dependsOn was set here https://github.com/dotnet/installer/blob/main/src/SourceBuild/content/eng/pipelines/ci.yml#:~:text=isBuiltFromVmr%3A%20true-,%24%7B%7B%20if%20and(ne(variables%5B%27System.TeamProject%27%5D%2C%20%27public%27)%2C%20notin(variables%5B%27Build.Reason,condition%3A%20always(),-Give%20feedback 
I removed it